### PR TITLE
Prohibit time overlap when editing segments in waveform

### PR DESCRIPTION
**Pull request by Dananji**
This fixes avalonmediasystem/avalon#3165

@adamjarling  I've written tests to all the scenarios I could come up with. I may have missed some. Please let me know if you can think of anymore test cases that can be added. Or anything that seems odd when you check this out :)

### CODE REVIEW

File: src/components/TimespanForm.js [hunk 1]
```diff
@@ -65,7 +67,28 @@ class TimespanForm extends Component {
...
+                {
+                  beginTime: structuralMetadataUtils.toHHmmss(startTime),
+                  endTime: structuralMetadataUtils.toHHmmss(endTime)
+                },
+                () => {
+                  this.updateChildOfOptions();
+                }
+              );
+            }
+          }
```
Comment (adamjarling) [inline src/components/TimespanForm.js]: Does this timing of the `setState` callbacks need to be this way?  Possible to use the `if (!this.formIsValid()) {` as the outer conditional maybe?   Just a thought...

Comment (Dananji) [reply src/components/TimespanForm.js]: Yeah, good point :+1: I will check it and see whether the functionality holds if use the condition outside.

Comment (Dananji) [reply src/components/TimespanForm.js]: @adamjarling It works even without the validity check. I removed the `if` condition totally, and pass the updated segment to `preventSegmentOverlapping()` function. And it returns valid values for the particular segment, when editing segment overlap other segments.

Comment (adamjarling) [reply src/components/TimespanForm.js]: @Dananji Great.

**Pull request merged**